internal/preferences.go

```diff
@@ -11,17 +11,22 @@ type InMemoryPreferences struct {
 	values          map[string]interface{}
 	lock            sync.RWMutex
 	changeListeners []func()
+	waitGroup       *sync.WaitGroup
 }
 
 // Declare conformity with Preferences interface
 var _ fyne.Preferences = (*InMemoryPreferences)(nil)
 
 // AddChangeListener allows code to be notified when some preferences change. This will fire on any update.
+// listener should not try to write values
 func (p *InMemoryPreferences) AddChangeListener(listener func()) {
 	p.lock.Lock()
 	defer p.lock.Unlock()
 
-	p.changeListeners = append(p.changeListeners, listener)
+	p.changeListeners = append(p.changeListeners, func() {
+		defer p.waitGroup.Done()
```

Comment:
I think this lacks symettery or proximity of paired operations.
If `p.waitGroup` is ever written, or if `p.changeListeners` ever has a different element added then `fireChange` will hang. I suggested an alternative below.